sklearn/naive_bayes.py

```diff
@@ -577,6 +577,16 @@ def _update_class_log_prior(self, class_prior=None):
             self.class_log_prior_ = np.full(n_classes, -np.log(n_classes))
 
     def _check_alpha(self):
+        # TODO(1.4): Remove
+        self._force_alpha = self.force_alpha
+        if self._force_alpha == "deprecated" or self._force_alpha is False:
+            self._force_alpha = False
+            warnings.warn(
+                "`force_alpha` was deprecated in 1.2 and will be removed in 1.4 with"
+                " models behaving as though it were set to `True`. To suppress this"
+                " warning, manually set the value of `force_alpha` to `True`.",
```

Comment:
Suppressing this warning will result in code that looks like:

```python
CategoricalNB(force_alpha=True)
```

which will end up breaking in v1.4 when `force_alpha` is removed.

I think we are forced to go through the 4 versions of having `force_alpha`. At first, we have `"force_alpha="warn"`, stating that the default will change from `False` to `True`. Then in v1.4 we switch it to `force_alpha="deprecated"` to state that `force_alpha` will be deprecated in v1.6.

Reply:
Ah I see. Will update to suit the double-deprecation soon -- thanks!